arch/common/stacktrace.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
+
+void __weak arch_stack_walk(stack_trace_callback_fn callback_fn, void *cookie,
```

Comment:
We are trying to get rid of weak symbols across the tree, especially for arch_ interfaces, can be done differently, something along the lines of the diff below, i.e. providing empty stubs instead.. 

```
--- a/arch/riscv/core/CMakeLists.txt
+++ b/arch/riscv/core/CMakeLists.txt
@@ -25,5 +25,5 @@ zephyr_library_sources_ifdef(CONFIG_RISCV_PMP pmp.c pmp.S)
 zephyr_library_sources_ifdef(CONFIG_THREAD_LOCAL_STORAGE tls.c)
 zephyr_library_sources_ifdef(CONFIG_USERSPACE userspace.S)
 zephyr_library_sources_ifdef(CONFIG_SEMIHOST semihost.c)
-zephyr_library_sources_ifdef(CONFIG_EXCEPTION_STACK_TRACE stacktrace.c)
+zephyr_library_sources(stacktrace.c)
 zephyr_linker_sources(ROM_START SORT_KEY 0x0vectors vector_table.ld)
diff --git a/arch/riscv/core/stacktrace.c b/arch/riscv/core/stacktrace.c
index 23baeca79d1..471de0b76f9 100644
--- a/arch/riscv/core/stacktrace.c
+++ b/arch/riscv/core/stacktrace.c
@@ -10,6 +10,7 @@
 #include <kernel_internal.h>
 #include <zephyr/logging/log.h>

+#if defined(CONFIG_EXCEPTION_STACK_TRACE)
 LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);

 uintptr_t z_riscv_get_sp_before_exc(const struct arch_esf *esf);
@@ -242,3 +243,10 @@ void z_riscv_unwind_stack(const struct arch_esf *esf, const _callee_saved_t *csf
        walk_stackframe(print_trace_address, &i, _current, esf, in_fatal_stack_bound, csf);
        LOG_ERR("");
 }
+#else
+void arch_stack_walk(stack_trace_callback_fn callback_fn, void *cookie,
+                    const struct k_thread *thread, const struct arch_esf *esf)
+{
+}
+
+#endif
```

Reply:
We can backport this if the first commit of #77067 is too much to be backported